src/edu/csus/ecs/pc2/ui/ShadowCompareRunsPane.java

```diff
@@ -303,10 +361,372 @@ public void actionPerformed(ActionEvent arg0) {
+
+                }
+            }
+        }
+        
+        //return the count of successfully updated runs
+        return numRunsUpdated;
+        
+    }
+
+    /**
+     * Invokes the PC2 server to update the status of the specified run to match the specified judgement.
+     * 
+     * @param submissionId the Id of the run to be updated.
+     * @param newJudgement the judgement which should be applied to the specified run.
+     * 
+     * @return true if the run was successfully updated; false if the run could not be updated for some reason 
+     *              (such as not being able to find the specified run, or the server failing to acknowledge a request
+     *              to edit/update the run within the server-timeout-limit).
+     */
+    protected boolean updateRun(Integer submissionId, CLICS_JUDGEMENT_ACRONYM newJudgement) {
+
+        log.log(Log.INFO, "Updating run " + submissionId + " to '" + newJudgement.name() + "'");
+        
+        //get all the runs
+        Run [] allRuns = getController().getContest().getRuns();
+        
+        //search for the desired run by Id
+        boolean found = false ;
+        Run targetRun = null;
+        for (Run run : allRuns) {
+            if (run.getNumber()==submissionId) {
+                targetRun = run;
+                found = true;
+                break;
+            }
+        }
+        
+        //if we didn't find the run, return failure
+        if (!found) {
+            
+            JOptionPane.showMessageDialog(this, "Failed to find submission " + submissionId, "Submission not found", JOptionPane.WARNING_MESSAGE);
+            log.log (Log.WARNING, "Failed to find run to be updated: " + submissionId);
+            return false;
+        }
+        
+        //if we get here we've found the run to be updated;
+        //try to find a PC2 Judgement that matches the remote CCS's CLICS judgement
+        Judgement [] judgementsArray = getContest().getJudgements();
+        for (Judgement judgement : judgementsArray) {
+            
+            if (newJudgement.name().contentEquals(judgement.getAcronym())) {
+                
+                //we found a matching PC2 judgement; check if the new judgement is a "yes"
+                boolean solved = CLICSJudgementType.isYesAcronym(newJudgement);
+                
+                //build a new JudgementRecord for PC2 containing the desired judgement values
+                JudgementRecord judgementRecord = new JudgementRecord(judgement.getElementId(), getContest().getClientId(), solved, false);
+
+                //duplicate the existing RunResultFiles, with null executionData (since we haven't actually re-executed the run)
+                RunResultFiles runResultFiles = new RunResultFiles(targetRun, targetRun.getProblemId(), judgementRecord, null);
+                
+                log.log(Log.INFO, "Sending new JudgementRecord to PC2 server: "
```

Comment:
This message does not show the run info/id update last statement
+ " isSolved=" + judgementRecord.isSolved() 
+ " run " + run);
If we were to actually spin off a thread for each update then we'd need to know which
run the JudgementRecord is for.

Reply:
Done.